R/ExtractData_Soils.R

```diff
@@ -263,6 +263,208 @@ extract_soil_CONUSSOIL <- function(MMC, sim_size, sim_space, dir_ex_soil,
+#'    }
+#'    \item Filenames expected to be in format 
+#'    \var{soiltype_M_sllayernumber_250m.tif}
+#'    Examples: \itemize{
+#'       \item Layer 2 clay = \var{clay_M_sl2_250m.tif}
+#'       \item Layer 4 bulk density = \var{bd_M_sl4_250m.tif}
+#'       }
+#'  }
+extract_soil_ISRIC250m <- function(MMC, sim_size, sim_space,
+                                       dir_ex_soil, fnames_in, resume, verbose,
+                                       default_TOC_GperKG = 0) {
+  # print stats
+  if (verbose) {
+    t1 <- Sys.time()
+    temp_call <- shQuote(match.call()[1])
+    print(paste0("rSFSW2's ", temp_call, ": started at ", t1))
+    on.exit({
+            print(paste0("rSFSW2's ", temp_call, ": ended after ",
+                          round(difftime(Sys.time(), t1, units = "secs"), 2),
+                          " s")) ; cat("\n")}, add = TRUE)
+  }
+  # set up data files for extraction =============================
+  # set temp variable show_site_info
+  show_site_info <- verbose
+  soil_layer <- 1
+  dir.ex.gridded <- file.path(dir_ex_soil, "ISRIC", "GriddedGlobalV5")
+  # stop program execution if folder path is incorrect
+  if (!dir.exists(dir.ex.gridded)) stop(paste0("Folder '", dir.ex.gridded,
+                                               "' does not exist"))
+  file_in_gridded <- list.files(dir.ex.gridded, pattern = ".tif$")
+  MMC[["source"]] <- "ISRIC_SoilGrids250m"
+  ldepth_gridded <- c(0, 5, 15, 30, 60, 100, 200) #in cm
+  todos <- is.na(MMC[["source"]]) | MMC[["source"]] == "ISRIC_SoilGrids250m"
+  stopifnot(requireNamespace("rgdal"))
+  # set the todo list to only extract data that doesn't already exist
+  if (resume) {
+    todos <- adjust_soils_todos(todos, MMC, sim_size)
+  }
+  names(todos) <- NULL
+  # get locations of simulation runs
+  sites_conus <- sim_space[["run_sites"]][todos, ]
+  # set this once now to avoid extracting data that already exists
+  n_extract <- sum(todos)
+  if (n_extract > 0) {
+    # main loop, does extraction for matricd, clay, sand, gravel, depth
+    for (tif in seq_along(file_in_gridded)) {
+      # get next tif file
+      tif_file <- file_in_gridded[tif]
+      # set soil types based on tif filenames
+      if (grepl("^bd", tif_file)) {
+        file_type <- "matricd"
+      } else if (grepl("^gravel", tif_file)) {
+        # switch file_type to match with MMC for insertion
+        file_type <- "GravelContent"
+      } else {
+        file_type <- substr(tif_file, 1, regexpr("_", tif_file) - 1)
+      }
+      # find the layer number of the current file ===================
+      ltemp <- strsplit(tif_file, "")
+      found_layer <- FALSE
+      i <- 1
+      while (!found_layer) {
+        if (grepl("[[:digit:]]", ltemp[[1]][i])) {
+          soil_layer <- type.convert(ltemp[[1]][i]) - 1
+          found_layer <- TRUE
+        } else {
+          i <- i + 1
+        }
+      }
+      # start extraction process =============================
+      
+        if (show_site_info) {
```

Comment:
I continue to believe that the placement of this print statement inside the for loop is a poor choice because it is invariant to the loop and necessitates unnecessary code such as `show_site_info <- FALSE`.